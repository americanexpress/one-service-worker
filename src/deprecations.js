import { deprecationNotice } from './utility/validation/deprecation';

import { createMetaCacheEntryName } from './cache';

// eslint-disable-next-line import/prefer-default-export
export const createCacheEntryName = deprecationNotice(createMetaCacheEntryName);
