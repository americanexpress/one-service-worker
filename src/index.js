/*
 * Copyright 2020 American Express Travel Related Services Company, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
 * or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import initialize from './initialization';

initialize();

export {
  on,
  once,
  off,
  emit,
  emitter,
  eventListeners,
  calls,
  createMiddleware,
  createMiddlewareContext,
  createMiddlewareFactory,
} from './utility/events';
export {
  TYPES,
  ENUMS,
  expectedType,
  expectedArrayOfType,
  enumerableException,
  unknownKey,
  unknownEventName,
  validateInput,
  getCacheOptions,
} from './utility/validation';
export {
  OneServiceWorkerError,
  errorFactory,
  notEnabled,
  notSupported,
  failedToInstall,
  failure,
} from './utility/errors';
export {
  configure,
  getConfig,
  isDevelopment,
  isEventsEnabled,
  isNonStandardEnabled,
  isNavigationPreloadEnabled,
  isServiceWorker,
  isServiceWorkerSupported,
  isNotificationSupported,
  isPushSupported,
  isBackgroundSyncSupported,
  isCacheStorageSupported,
  isIndexedDBSupported,
  isPermissionsSupported,
  isOffline,
} from './utility/runtime';

export {
  getRegistration,
  getRegistrations,
  register,
  getWorker,
  postMessage,
  update,
  unregister,
  escapeHatch,
  showNotification,
  getNotifications,
  getSubscription,
  subscribe,
  unsubscribe,
  registerTag,
  getTags,
  urlBase64ToUint8Array,
} from './core';

export {
  defaultCacheName,
  defaultCacheOptions,
  normalizeRequest,
  open,
  has,
  keys,
  match,
  matchAll,
  add,
  addAll,
  put,
  remove,
  removeAll,
  entries,
  clear,
  metaDataCacheName,
  createCacheName,
  createCacheEntryName,
  createMetaRequest,
  createMetaResponse,
  getMetaStore,
  getMetaData,
  setMetaData,
  deleteMetaData,
} from './cache';

export {
  onInstall,
  onActivate,
  onMessage,
  onPush,
  onSync,
  onFetch,
  cacheBusting,
  cacheRouter,
  cacheStrategy,
  clientsClaim,
  escapeHatchRoute,
  manifest,
  messageContext,
  messenger,
  navigationPreloadActivation,
  navigationPreloadResponse,
  appShell,
  precache,
  skipWaiting,
  expiration,
} from './middleware';
