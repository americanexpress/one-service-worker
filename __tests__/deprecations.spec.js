import * as deprecatedAPI from '../src/deprecations';
import { printExports } from './helpers';

describe('deprecations', () => {
  test('consistently exports API items to be deprecated by next major release', () => {
    expect.assertions(1);
    expect(printExports(deprecatedAPI)).toMatchSnapshot();
  });
});
