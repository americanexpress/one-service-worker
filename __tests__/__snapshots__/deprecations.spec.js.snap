// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deprecations consistently exports API items to be deprecated by next major release 1`] = `"createCacheEntryName :: function"`;
